src/deepsparse/kv_cache.py

```diff
@@ -0,0 +1,37 @@
+#############################################################################
+# This is a placeholder class to be replaced by LIB.deepsparse_engine.KVCache
+#############################################################################
+
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, List
+
+import numpy
+
+
+class KVCache:
+    def __init__(
+        self,
+        cache: Dict[str, numpy.ndarray],
+        prompt_length: int,
```

Comment:
Why do we need to know the prompt_length in advance? To allocate the buffers for the kv cache ahead of time?

Reply:
Oh i see this is not as much as prompt_length, rather the "tokens seen so far" right?

Reply:
This data structure is passed into the engine at runtime. Prompt_length is necessary here so that the engine knows how many tokens were used to generate the KV cache since it will be sized internally based on the context length. The prompt length will be 0 for cases where we are building up the KV cache from scratch using a single token model. I called it prompt_length because we typically describe the list of tokens passed into the multi_token model as a prompt. I'm open to suggestions W.R.T naming, though.